imap_processing/lo/l0/science_direct_events.py

```diff
@@ -85,116 +85,155 @@ class ScienceDirectEvents(LoL0):
     TIME: float
     ENERGY: float
     POS: float
-    case_number: int
-    tof_calculation_binary: dict
-    tof_decoder: list
-    remaining_bits: dict
-    parsed_bits: list
 
     def __init__(self, packet, software_version: str, packet_file_name: str):
+        """Intialization method for Science Direct Events Data class."""
         super().__init__(software_version, packet_file_name, CcsdsData(packet.header))
         self.parse_data(packet)
+        self._decompress_data()
 
-    def decompress_data(self):
+    def _decompress_data(self):
         """Decompress the Lo Science Direct Events data."""
-        self._find_decompression_case()
-        self._find_tof_decoder_for_case()
-        self._read_tof_calculation_table()
-        self._find_remaining_bits()
-        self._parse_binary()
-        self._decode_fields()
-
-    def _find_tof_decoder_for_case(self):
-        """Get the TOF decoder for this DE's case number."""
-        # Case 0 can either be a gold or silver triple. Gold triples do
-        # not send down the TOF1 value and instead recover the TOF1 value
-        # on the ground using the decompressed checksum
-        if self.case_number == 0:
+        case_number = self._find_decompression_case()
+        tof_decoder = self._find_tof_decoder_for_case(case_number)
+        tof_calculation_binary = self._read_tof_calculation_table(case_number)
+        remaining_bit_coefficients = self._find_remaining_bit_coefficients(
+            tof_calculation_binary
+        )
+        parsed_bits = self._parse_binary(case_number, tof_decoder)
+        # decode_fields will set the TOF class variables
+        self._decode_fields(remaining_bit_coefficients, parsed_bits)
+
+    def _find_decompression_case(self):
+        """Find the decompression case for this DE.
+
+        The first 4 bits of the binary data are used to
+        determine which case number we are working with.
+        The case number is used to determine how to
+        decompress the TOF values.
+        """
+        return int(self.DATA[0:4].bin, 2)
+
+    def _find_tof_decoder_for_case(self, case_number):
+        """Get the TOF decoder for this DE's case number.
+
+        The case number determines wich TOF decoder to use.
+        The TOF decoder table shows how the TOF bits should be
+        parsed in the binary data.
+
+        Case 0 can either be a gold or silver triple. Gold triples do
+        not send down the TOF1 value and instead recover the TOF1 value
+        on the ground using the decompressed checksum.
+        Cases 4, 6, 10, 12, 13 may be Bronze triples. If it's not a bronze triple,
+        the Position is not transmitted, but TOF3 is. If it is bronze, the table
+        should be used as is. If it's not bronze, position was not transmitted,
+        but TOF3 was transmitted.
```

Comment:
Wow, this is complicated! But, I think your explanation and the structure of your code line up nicely